test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewFeatures/AttributeDictionaryTest.cs

```diff
@@ -0,0 +1,432 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.ViewFeatures
+{
+    public class AttributeDictionaryTest
+    {
+        [Fact]
+        public void AttributeDictionary_AddItems()
+        {
+            // Arrange
+            var attributes = new AttributeDictionary();
+
+            // Act
+            attributes.Add("zero", "0");
+            attributes.Add("one", "1");
+            attributes.Add("two", "2");
+
+            // Assert
+            Assert.Equal(3, attributes.Count);
+            Assert.Collection(
+                attributes,
+                kvp => Assert.Equal(new KeyValuePair<string, string>("one", "1"), kvp),
+                kvp => Assert.Equal(new KeyValuePair<string, string>("two", "2"), kvp),
+                kvp => Assert.Equal(new KeyValuePair<string, string>("zero", "0"), kvp));
+        }
+
+        [Fact]
+        public void AttributeDictionary_AddItems_AsKeyValuePairs()
+        {
+            // Arrange
+            var attributes = new AttributeDictionary();
+
+            // Act
+            attributes.Add(new KeyValuePair<string, string>("zero", "0"));
+            attributes.Add(new KeyValuePair<string, string>("one", "1"));
+            attributes.Add(new KeyValuePair<string, string>("two", "2"));
+
+            // Assert
+            Assert.Equal(3, attributes.Count);
+            Assert.Collection(
+                attributes,
+                kvp => Assert.Equal(new KeyValuePair<string, string>("one", "1"), kvp),
+                kvp => Assert.Equal(new KeyValuePair<string, string>("two", "2"), kvp),
+                kvp => Assert.Equal(new KeyValuePair<string, string>("zero", "0"), kvp));
+        }
+
+        [Fact]
+        public void AttributeDictionary_Add_DuplicateKey()
+        {
+            // Arrange
+            var attributes = new AttributeDictionary();
+
+            attributes.Add("zero", "0");
+            attributes.Add("one", "1");
+            attributes.Add("two", "2");
+
+            // Act & Assert
+            Assert.Throws<InvalidOperationException>(() => attributes.Add("one", "15"));
+        }
+
+        [Fact]
+        public void AttributeDictionary_Clear()
+        {
+            // Arrange
+            var attributes = new AttributeDictionary();
+
+            attributes.Add(new KeyValuePair<string, string>("zero", "0"));
+            attributes.Add(new KeyValuePair<string, string>("one", "1"));
+            attributes.Add(new KeyValuePair<string, string>("two", "2"));
+
+            // Act
+            attributes.Clear();
+
+            // Assert
+            Assert.Equal(0, attributes.Count);
+            Assert.Empty(attributes);
+        }
+
+        [Fact]
+        public void AttributeDictionary_Contains()
```

Comment:
Can you add negative cases for the `Contains` \ `ContainsKeys` (the `Assert.False` scenarios?)